components/salsa-2022-macros/src/salsa_struct.rs

```diff
@@ -89,6 +99,14 @@ impl<A: AllowedOptions> SalsaStruct<A> {
         self.fields.iter()
     }
 
+    pub(crate) fn is_identity_field(&self, field: &SalsaField) -> bool {
+        match self.kind {
+            SalsaStructKind::Input => false,
```

Comment:
This is correct, but it's going to be annoying in practice and I'm not sure what to do about it. :/

I'm imagining that e.g. you have `InputFile { path: String, contents: String }`, and you would like to include `path` when you print out the input-file, so you don't just see `InputFile(3)`. 

I wonder if it makes sense to have the ability to mark input fields as `#[id]` and then include them in the printouts (and maybe even make them immutable?). It seems like it might be confusing to do that since, in fact, they are not part of the "identity" of the input.

Reply:
Maybe the method can be renamed to `is_stable_field`, `is_immutable_field` or `is_non_changing_field` and another attribute can be added like `#[stable]`, `#[immutable]` or `#[meta]`.

Reply:
Why don't we file a bug about this problem. I think we need some way to tag fields in a `salsa::input` and have them show up, and I think those fields should be immutable. The `#[id]` attribute feels nice to me in that we already use it on `salsa::tracked` for a "similar in spirit" sort of meaning -- if this field were to change, it'd be a different struct, so to speak. So e.g. I can imagine an `InputFile` might mark the path as `#[id]` to indicate that it just doesn't change (and if it did, we'd make a new input).

But it's also confusing in that, unlike with tracked structs, there is no "id" in the same way.

Reply:
Filed https://github.com/salsa-rs/salsa/issues/410